crates/red_knot/src/lint.rs

```diff
@@ -140,6 +142,66 @@ fn lint_unresolved_imports(context: &SemanticLintContext) -> QueryResult<()> {
     Ok(())
 }
 
+fn lint_bad_overrides(context: &SemanticLintContext) -> QueryResult<()> {
+    // TODO possibly we should have a special marker on the real typing module (from typeshed) so
+    // if you have your own "typing" module in your project, we don't consider it THE typing module
+    let Some(typing_module) = context.db.resolve_module(ModuleName::new("typing"))? else {
+        debug_assert!(false, "typing module should always exist");
+        return Ok(());
+    };
+    let typing_file = context.db.module_to_file(typing_module)?;
+    let typing_table = context.db.symbol_table(typing_file)?;
```

Comment:
Yes, I think that's a good direction to move the API.

I want to avoid over-fitting our API to our first few lint rules, so it was kind of intentional (also just that I was short on time) that I didn't do much API design thinking in this PR, just pushed to get it working. I want to abstract things when it is clear we have multiple needs for them. But I think this is a pretty clear case of something we will have many uses for.

Reply:
yeah, that sounds reasonable to me. It might be an opportunity for Patrick to work out an API for this as part of the work he does on the new lint rule.

Reply:
I added a `resolve_global_symbol` query to encapsulate this.